### PR TITLE
Introduce delayed sampling mechanism

### PR DESCRIPTION
**Pull request by lahead**
This change introduces a mechanism called "delayed sampling" which aims at minimizing the CPU overhead related to output tokens post-processing and next token scheduling time by overlapping the CPU-active part with device computations.

When delayed sampling is enabled first prompt model execution schedules the model.forward() and logits computation on the device followed by immediately returning an output filled with invalid token ids, not waiting for the computation to complete. The output logits are only gathered and sampled in the subsequent model execution, which again schedules next model.forward() and logits computation invocation not waiting for the results to come back, but rather returning the previously collected and sampled output token ids. This process continues for the entire sequence length resulting in the last token which is computed redundantly being discarded.

Please review @madamczykhabana , @kzawora-intel

### CODE REVIEW

**Review by madamczykhabana — APPROVED** [review]
LGTM
